packages/next-swc/crates/core/tests/fixture.rs

```diff
@@ -260,3 +271,50 @@ fn shake_exports_fixture_default(input: PathBuf) {
         &output,
     );
 }
+
+#[fixture("tests/fixture/emotion/*/input.tsx")]
+fn next_emotion_fixture(input: PathBuf) {
+    let output = input.parent().unwrap().join("output.ts");
+    let mut comments: Rc<SingleThreadedComments> = Default::default();
+    let comments_raw = &mut comments as *mut Rc<SingleThreadedComments>;
+    test_fixture(
+        ts_syntax(),
+        &|tr| {
```

Comment:
@kdy1 I need `FnMut` here to pass `&tr.comments` into `jsx` and `emotion`

Reply:
BTW, the `SingleThreadedComments` in `tr` seems not to work, no `comments` are emitted in testing.

Reply:
Only `/*#__PURE__*/` for tree-shaking

Reply:
The other way is to use 

https://github.com/vercel/next.js/blob/canary/packages/next-swc/crates/core/tests/full.rs

If you do it you will need to deserialize `config.json` as TransformOptions

Reply:
Ah, you can clone the `tr.comments`

Reply:
Why do you use your own comments, instead of cloning it of `Tester`?

Reply:
> Why do you use your own comments, instead of cloning it of Tester?

Because `emotion` only accepts `&comments`, the `Compiler::comments` return the `&comments`.
`Clone` `tr.commens` here will be dropped out of the `Fn`. I need another `comments` out of the `Fn` scope to make `&comments` life time longer than `emotion`.

Reply:
`Comments` is cheap to clone, and you can simply clone it.